pysatMadrigal/instruments/gnss_tec.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-.
+"""Supports the MIT Haystack GNSS TEC data products
+
+The Global Navigation Satellite System (GNSS) is used in conjunction with a
+world-winde receiver network to produce total electron content (TEC) data
+products, including vertical and line-of-sight TEC.
+
+Downloads data from the MIT Haystack Madrigal Database.
+
+Properties
+----------
+platform
+    'gnss'
+name
+    'tec'
+tag
+    'vtec', 'los'
+
+Examples
+--------
+::
+
+    import datetime
+    import pysat
+    import pysatMadrigal as pymad
+
+    vtec = pysat.Instrument(inst_module=pymad.instruments.gnss_tec, tag='vtec')
+    vtec.download(dt.datetime(2017, 11, 19), dt.datetime(2017, 11, 20),
+                  user='Firstname+Lastname', password='email@address.com')
+    vtec.load(date=dt.datetime(2017, 11, 19))
+
+
+Note
+----
+    Please provide name and email when downloading data with this routine.
+
+"""
+
+import datetime as dt
+import functools
+import numpy as np
+
+from pysat.instruments.methods import general as ps_gen
+
+from pysatMadrigal.instruments.methods import madrigal as mad_meth
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+platform = 'gnss'
+name = 'tec'
+tags = {'vtec': 'vertical TEC', 'los': 'slant TEC'}
+sat_ids = {'': list(tags.keys())}
+_test_dates = {'': {'vtec': dt.datetime(2017, 11, 19),
+                    'los': dt.datetime(2017, 11, 19)}}
+pandas_format = False
+
+# support list files routine
+# use the default pysat method
+dname = '{year:4d}{month:02d}{day:02d}'
+vname = '.{version:03d}'
+supported_tags = {ss: {'vtec': "gps{:s}g{:s}.hdf5".format(dname, vname),
+                       'los': "los_{:s}{:s}.h5".format(dname, vname)}
+                  for ss in sat_ids.keys()}
+list_files = functools.partial(ps_gen.list_files,
+                               supported_tags=supported_tags)
+
+# madrigal tags
+madrigal_inst_code = 8000
+madrigal_tag = {'': {'vtec': 3500, 'los': 3505}}
+
+# support listing files currently available on remote server (Madrigal)
+list_remote_files = functools.partial(mad_meth.list_remote_files,
+                                      supported_tags=supported_tags,
+                                      inst_code=madrigal_inst_code)
+
+
+def init(self):
+    """Initializes the Instrument object with values specific to JRO ISR
+
+    Runs once upon instantiation.
+
+    Parameters
+    ----------
+    self : pysat.Instrument
+        This object
+
+    """
+
+    ackn_str = ' '.join(["GPS TEC data products and access through the ",
```

Comment:
There are spaces included in the text as well as a forced space ' ' with the join.